Modules/_bz2module.c

```diff
@@ -602,16 +641,23 @@ _bz2_BZ2Decompressor_decompress_impl(BZ2Decompressor *self, Py_buffer *data,
 }
 
 /*[clinic input]
-_bz2.BZ2Decompressor.__init__
-
-Create a decompressor object for decompressing data incrementally.
+_bz2.BZ2Decompressor.__reduce__
 
-For one-shot decompression, use the decompress() function instead.
 [clinic start generated code]*/
 
+static PyObject *
+_bz2_BZ2Decompressor___reduce___impl(BZ2Decompressor *self)
+/*[clinic end generated code: output=f6a40650813f482e input=8db9175a609fdd43]*/
+{
+    PyErr_Format(PyExc_TypeError,
+                 "cannot pickle %s object",
+                 Py_TYPE(self)->tp_name);
+    return NULL;
+}
+
+// TODO: Convert it to clinic
```

Comment:
Maybe explain why Argument Clinic is not used. I understand that the issue is that AC always wants to check the type which would be wrong here.